cdap-api/src/main/java/co/cask/cdap/api/lineage/Operation.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.lineage;
+
+/**
+ * Abstract base class to represent Operation with name and description.
+ */
+public abstract class Operation {
+  private final String name;
+  private final String description;
+
+  protected Operation(String name, String description) {
+    this.name = name;
+    this.description = description;
+  }
+
+  /**
+   * @return the name of the operation, typically its "READ"
+   */
+  public String getName() {
+    return name;
+  }
+
+  /**
+   * @return the description associated with the operation
+   */
+  public String getDescription() {
+    return description;
+  }
+
+  /**
+   * Builder for Operation
+   */
+  public abstract static class Builder {
+    protected final String name;
+    protected String description;
+
+    public Builder(String name) {
+      this.name = name;
+    }
+
+    /**
+     * Set description for the READ operation.
```

Comment:
not READ

Reply:
Fixed. was left-over from copy. Btw renamed this class to `AbstractOperation`.